runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -6492,6 +6492,274 @@ static TR::Register* inlineIntrinsicIndexOf(TR::Node* node, TR::CodeGenerator* c
+   TR_ASSERT_FATAL_WITH_NODE(node, node->getNumChildren() == 5, "Wrong number of children in inlineStringLatin1Inflate");
+
+   TR::Node *srcNode = node->getFirstChild();
+   TR::Node *srcOffsetNode = node->getSecondChild();
+   TR::Node *dstNode = node->getThirdChild();
+   TR::Node *dstOffsetNode = node->getChild(3);
+   TR::Node *lengthNode = node->getChild(4);
+
+   TR_ARM64ScratchRegisterManager *srm = cg->generateScratchRegisterManager();
+   TR::Register *srcReg = cg->evaluate(srcNode);
+   TR::Register *srcAddrReg = (srcNode->getReferenceCount() > 1) ? srm->findOrCreateScratchRegister() : srcReg;
+   const bool isSrcOffsetConstZero = srcOffsetNode->isConstZeroValue();
+   TR::Register *srcOffsetReg = isSrcOffsetConstZero ? NULL : cg->evaluate(srcOffsetNode);
+   TR::Register *dstReg = cg->evaluate(dstNode);
+   TR::Register *dstAddrReg = (dstNode->getReferenceCount() > 1) ? srm->findOrCreateScratchRegister() : dstReg;
+   const bool isDstOffsetConstZero = dstOffsetNode->isConstZeroValue();
+   TR::Register *dstOffsetReg = isDstOffsetConstZero ? NULL : cg->evaluate(dstOffsetNode);
+   TR::Register *savedLengthReg = cg->evaluate(lengthNode);
+
+   TR::LabelSymbol *doneLabel = generateLabelSymbol(cg);
+   TR_Debug *debugObj = cg->getDebug();
+
+   auto branchToDoneIfZeroInstr = generateCompareBranchInstruction(cg, TR::InstOpCode::cbzw, node, savedLengthReg, doneLabel);
+   if (debugObj)
+      {
+      debugObj->addInstructionComment(branchToDoneIfZeroInstr, "Branch to doneLabel if the length is 0");
+      }
+
+   generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addimmx, node, srcAddrReg, srcReg, TR::Compiler->om.contiguousArrayHeaderSizeInBytes());
+   generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addimmx, node, dstAddrReg, dstReg, TR::Compiler->om.contiguousArrayHeaderSizeInBytes());
+   if (!isSrcOffsetConstZero)
+      {
+      generateTrg1Src2Instruction(cg, TR::InstOpCode::addx, node, srcAddrReg, srcAddrReg, srcOffsetReg);
+      }
+   if (!isDstOffsetConstZero)
+      {
+      generateTrg1Src2ShiftedInstruction(cg, TR::InstOpCode::addx, node, dstAddrReg, dstAddrReg, dstOffsetReg, TR::SH_LSL, 1);
+      }
+   TR::Register *lengthReg = (lengthNode->getReferenceCount() > 1) ? srm->findOrCreateScratchRegister() : savedLengthReg;
+   if (lengthNode->getReferenceCount() > 1)
+      {
+      generateMovInstruction(cg, node, lengthReg, savedLengthReg);
+      }
+   TR::LabelSymbol *GEQ8BytesLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *GEQ4BytesLabel = NULL;
+   TR::LabelSymbol *GEQ2BytesLabel = NULL;
+   TR::Register *vtmp0Reg = srm->findOrCreateScratchRegister(TR_VRF);
+   TR::Register *vtmp1Reg = srm->findOrCreateScratchRegister(TR_VRF);
+   TR::Register *srcEndReg = NULL;
+   TR::Register *dstEndReg = NULL;
+
+   GEQ4BytesLabel = generateLabelSymbol(cg);
+   GEQ2BytesLabel = generateLabelSymbol(cg);
```

Comment:
Is there any reason for generating these labels here instead of L.6605-6606?

Reply:
Thanks for pointing it out. I updated the line 6605 and 6606 to generate labels.